assets/model_monitoring/components/src/shared_utilities/run_metrics_utils.py

```diff
@@ -136,7 +139,6 @@ def publish_metrics(run_id: str, metrics: dict, step: int):
             try:
                 mlflow.log_metrics(metrics=metrics, step=step)
                 return
-            except Exception as e:
-                print(f"Exception occurred in publishing metrics: {traceback.format_exc()}")
+            except Exception:
```

Comment:
if the run fail because update status from failed to completed, seems we can also catch the exception here? If so, we don't need the outer part of try-except.

Reply:
As we discussed, if we have this logic at the beginning, we should be good.
The reason we need to add outside, it's because there is already some failed run

Reply:
yes, there are some failed run there, so the mlfow.log_metrics may fail because update run status from failed to completed, but seems this exception will be also caught here and be silence ignored after 3 retries, and continue with next metrics run outside, do we still need to catch exception outside?